packages/esm-patient-search-app/src/patient-search-page/refine-search/refine-search-tablet.component.tsx

```diff
@@ -0,0 +1,182 @@
+  const renderSearchFields = () => {
+    const fields: Array<SearchFieldConfig> = [];
+
+    Object.entries(config.search.searchFields.fields).forEach(([fieldName, fieldConfig]) => {
+      if (fieldConfig.enabled) {
+        fields.push({
+          name: fieldName,
+          type: fieldName as SearchFieldType,
+          label: fieldConfig.label,
+        });
+      }
+    });
+
+    config.search.searchFields.personAttributes?.forEach((attribute: PersonAttributeFieldConfig) => {
+      fields.push({
+        name: attribute.attributeTypeUuid,
+        type: 'personAttribute',
+        label: attribute.label,
+        attributeTypeUuid: attribute.attributeTypeUuid,
+        ...attribute,
+      });
+    });
+
+    const genderField = fields.find((field) => field.type === 'gender');
+    const dobField = fields.find((field) => field.type === 'dateOfBirth');
+    const otherFields = fields.filter((field) => !['gender', 'dateOfBirth'].includes(field.type));
+
+    const renderOtherFields = () => {
+      const rows: SearchFieldConfig[][] = [];
+      for (let i = 0; i < otherFields.length; i += 3) {
+        rows.push(otherFields.slice(i, i + 3));
+      }
+
+      return rows.map((row, index) => (
+        <div key={index} className={styles.otherFieldsRow}>
+          {row.map((field) => (
+            <div key={field.name} className={styles.fieldTabletOrOverlay}>
+              <SearchField
+                field={field}
+                formState={formState}
+                inTabletOrOverlay={true}
+                isTablet={isTablet}
+                onInputChange={onInputChange}
+                onDateOfBirthChange={onDateOfBirthChange}
+              />
+            </div>
+          ))}
+        </div>
+      ));
+    };
+
+    return (
+      <>
+        {(genderField || dobField) && (
+          <div className={classNames(styles.padded, styles.refineSearchDialogGenderSexRow)}>
+            {genderField && (
+              <div className={styles.fieldTabletOrOverlay}>
+                <SearchField
+                  field={genderField}
+                  formState={formState}
+                  inTabletOrOverlay={true}
+                  isTablet={isTablet}
+                  onInputChange={onInputChange}
+                  onDateOfBirthChange={onDateOfBirthChange}
+                />
+              </div>
+            )}
+            {dobField && (
+              <div className={styles.fieldTabletOrOverlay}>
+                <SearchField
+                  field={dobField}
+                  formState={formState}
+                  inTabletOrOverlay={true}
+                  isTablet={isTablet}
+                  onInputChange={onInputChange}
+                  onDateOfBirthChange={onDateOfBirthChange}
+                />
+              </div>
+            )}
+          </div>
+        )}
+        {otherFields.length > 0 && (
+          <div className={classNames(styles.padded, styles.otherFieldsContainer)}>{renderOtherFields()}</div>
```

Comment:
Better not to call this as a function with no params, but rather render the variable.

Reply:
You can make the Memo